cas-server-webapp-validation/src/main/java/org/jasig/cas/web/AbstractServiceValidateController.java

```diff
@@ -128,13 +151,38 @@ protected Credential getServiceCredentialsFromRequest(final WebApplicationServic
                 verifyRegisteredServiceProperties(registeredService, service);
                 return new HttpBasedServiceCredential(new URL(pgtUrl), registeredService);
             } catch (final Exception e) {
-                logger.error("Error constructing pgtUrl", e);
+                logger.error("Error constructing {}", CasProtocolConstants.PARAMETER_PROXY_CALLBACK_URL, e);
             }
         }
 
         return null;
     }
 
+
+    /**
+     * Validate authentication context pair.
+     *
+     * @param assertion the assertion
+     * @param request   the request
+     * @return the pair
+     */
+    protected Pair<Boolean, Optional<MultifactorAuthenticationProvider>> validateAuthenticationContext(
+            final Assertion assertion, final HttpServletRequest request) {
+        final String requestedContext = request.getParameter(this.authenticationContextParameter);
```

Comment:
yeah, that's the premise. I want to make sure the trigger of "Opt-In Request Parameter" can still be enforced at validation by still supporting `requestContext` in that scenario.

I'll make the change.